h2/src/main/org/h2/command/Parser.java

```diff
@@ -4295,15 +4295,19 @@ private Expression readBuiltinFunctionIf(String upperName) {
         case "DIFFERENCE":
             return new SoundexFunction(readExpression(), readLastArgument(), SoundexFunction.DIFFERENCE);
         case "JSON_OBJECT": {
+            boolean acceptsCommaAsValueSeparator = database.getMode().getEnum() == ModeEnum.MariaDB ||
+                    database.getMode().getEnum() == ModeEnum.MySQL;
             JsonConstructorFunction function = new JsonConstructorFunction(false);
             if (currentTokenType != CLOSE_PAREN && !readJsonObjectFunctionFlags(function, false)) {
                 do {
                     boolean withKey = readIf(KEY);
                     function.addParameter(readExpression());
                     if (withKey) {
                         read(VALUE);
-                    } else if (!readIf(VALUE)) {
-                        read(COLON);
+                    } else {
+                        if (!(readIf(VALUE) || (acceptsCommaAsValueSeparator && readIf(COMMA)))) {
+                            read(COLON);
+                        }
```

Comment:
`JSON_OBJECTAGG` needs the same change too, MySQL and MariaDB also use their own syntax for them:
https://dev.mysql.com/doc/refman/8.0/en/aggregate-functions.html#function_json-objectagg
https://mariadb.com/kb/en/json_objectagg/

https://github.com/h2database/h2database/blob/d05cb83783552cc022b7fad1d965ee06bd5e6149/h2/src/main/org/h2/command/Parser.java#L3574-L3581